indexing-service/src/main/java/org/apache/druid/indexing/common/task/CompactionTask.java

```diff
@@ -422,9 +423,40 @@ public boolean isPerfectRollup()
     return tuningConfig != null && tuningConfig.isForceGuaranteedRollup();
   }
 
+  @VisibleForTesting
+  void emitCompactIngestionModeMetrics(
+      ServiceEmitter emitter,
+      boolean isDropExisting
+  )
+  {
+
+    if (emitter == null) {
+      return;
+    }
+
+    BatchIngestionMode mode = getBatchIngestionMode(
+        false, // compact never supports append mode
+        isDropExisting
+    );
+    // compact does not support APPEND
+    switch (mode) {
+      case REPLACE:
+        emitter.emit(buildEvent("compact/replace/count", 1));
+        break;
+      case OVERWRITE:
+        emitter.emit(buildEvent("compact/overwrite/count", 1));
+        break;
+      default:
+        throw new ISE("Invalid compact ingestion mode [%s]", mode);
```

Comment:
Should we just log error instead of possibly failing ingestion task with this thrown exception?

Reply:
Added an INVALID mode and logging a warning instead when mode is INVALID.